tests/integration/worker/test_submission_worker.py

```diff
@@ -0,0 +1,394 @@
+        )
+
+        mock_lb.assert_not_called()
+
+        self.assertEqual(self.submission.started_at, starting_time)
+        self.assertEqual(self.submission.status, Submission.FAILED)
+        self.assertEqual(self.submission.completed_at, ending_time)
+
+        expected_output = {"result": ""}
+        self.assertEqual(self.submission.output, expected_output)
+
+        mock_shutil.rmtree.assert_called_with(temp_run_dir)
+
+    def test_run_submission_when_challenge_phase_split_does_not_exist(self, mock_map, mock_script_dict,
+                                                                      mock_createdir, mock_lb,
+                                                                      mock_shutil, mock_timezone,
+                                                                      mock_open, mock_cf):
+        challenge_id = self.challenge.id
+        phase_id = self.challenge_phase.id
+        user_annotation_file_path = "tests/integration/worker/data/user_annotation.txt"
+        temp_run_dir = "mocked/dir/submission_{}/run".format(self.submission.id)
+
+        mock_map[challenge_id] = mock.Mock()
+        mock_map.get(challenge_id).get.return_value = "test_annotation_file.txt"
+        mock_script_dict[challenge_id] = mock.Mock()
+        mock_script_dict[challenge_id].evaluate.return_value = {"result": [{"split2": {"metric1": 10}}]}
+
+        starting_time = timezone.now()
+        time.sleep(0.5)
+        ending_time = timezone.now()
+        mock_timezone.now.side_effect = [starting_time, ending_time]
+
+        if not os.path.exists(temp_run_dir):
+            os.makedirs(temp_run_dir)
+
+        patcher = mock.patch("scripts.workers.submission_worker.ContentFile")
+        mock_cf = patcher.start()
+        mock_cf.return_value = ContentFile("")
+
+        submission_worker.run_submission(challenge_id, self.challenge_phase, self.submission, user_annotation_file_path)
+
+        self.assertEqual(mock_open.return_value.write.call_args_list[0],
+                         mock.call("ORGINIAL EXCEPTION: No such relation between Challenge Phase and DatasetSplit"
+                                   " specified by Challenge Host \n"))
+
+        annotation_file_path = "mocked/dir/challenge_data/challenge_{}/phase_data/phase_{}/test_annotation_file.txt".format(
+            challenge_id, phase_id)
+
+        mock_createdir.assert_called_with(temp_run_dir)
+
+        mock_script_dict[challenge_id].evaluate.assert_called_with(
+            annotation_file_path,
+            user_annotation_file_path,
+            self.challenge_phase.codename,
+            submission_metadata="",
+        )
+
+        mock_lb.objects.bulk_create.assert_not_called()
+
+        self.assertEqual(self.submission.started_at, starting_time)
+        self.assertEqual(self.submission.status, Submission.FAILED)
+        self.assertEqual(self.submission.completed_at, ending_time)
+
+        expected_output = {"result": [{"split2": {"metric1": 10}}]}
```

Comment:
How is this an expected output when `challenge_phase_split` doesn't exist? It should throw an error and break as shown here: https://github.com/Cloud-CV/EvalAI/blob/master/scripts/workers/submission_worker.py#L446

Reply:
The worker is structured such that anything inside the `submission_output`'s result key is saved as output irrespective of what's inside. So, to avoid saving any output in such a case, the `submission_output` variable could be changed to False in the except block. 
Then these lines won't execute: (https://github.com/Cloud-CV/EvalAI/blob/46ac6539837b3b7da8951fdebe95811e445fab2c/scripts/workers/submission_worker.py#L510)
```
if submission_output:
        output = {}
        output["result"] = submission_output.get("result", "")
        submission.output = output
```
I'll make the required changes.

Reply:
`submission_output` is a dict, so it can't be set to False. We can have an additional condition in this if condition to check if the submission is successful or not and if successful then we should save the output and rest other stderr and stdout fields should remain as is it. The check should be while saving the output.

Reply:
Please don't incorporate that change in this PR. Let's get this merged and please open up a new PR for that along with an issue.

Reply:
> submission_output is a dict, so it can't be set to False

It can actually, I've tried it. There's no restriction as well, and `submission_output` is not used again. Another way is to set it to an empty dict (will be evaluated to False). But an additional condition with the succesful_submission flag is cleaner.

> We can have an additional condition in this if condition to check if the submission is successful or not and if successful then we should save the output and rest other stderr and stdout fields should remain as is it. The check should be while saving the output.

Sure.